js/src/components/tree-select-control/index.js

```diff
@@ -100,6 +112,67 @@ const TreeSelectControl = ( {
 		setTreeVisible( false );
 	} );
 
+	/**
+	 * Perform the search query filter in the Tree options
+	 *
+	 * 1. Check if the search query is already cached and return it if so.
+	 * 2. Deep Copy the tree. Since we are going to modify his children and labels recursively.
```

Comment:
💅 `his` looks like a typo?

```suggestion
	 * 2. Deep Copy the tree. Since we are going to modify its children and labels recursively.
```